ArcGISRuntimeSDKQt_QMLSamples/Layers/ExportVectorTiles/ExportVectorTiles.qml

```diff
@@ -0,0 +1,275 @@
+
+            Text {
+                id: progressText
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: exportProgress + "% Completed"
+                font.pixelSize: 16
+            }
+
+            onWidthChanged: {
+                exportProgressWindow.width = exportProgressColumn.width + 20
+            }
+
+            onHeightChanged: {
+                exportProgressWindow.height = exportProgressColumn.height + 20
+            }
+        }
+    }
+
+    // Create an extent rectangle for selecting the offline area
+    Rectangle {
+        id: extentRectangle
+        anchors.centerIn: parent
+
+        width: parent.width - parent.width * 0.1
+        height: parent.height - parent.height * 0.25
+        color: "transparent"
+        border {
+            color: "red"
+            width: 3
+        }
+
+        visible: exportJobStatus === 0 || exportJobStatus === 4
+    }
+
+    // Button to start the download
+    Button {
+        id: button
+        anchors {
+            bottom: parent.bottom
+            bottomMargin: rootRectangle.height * .05
+            horizontalCenter: parent.horizontalCenter
+        }
+        width: 150
+
+        text: "Export area"
+
+        onClicked: {
+            switch(exportJobStatus) {
+            case 0: // Not started
+                startExport(extentRectangle.x, (extentRectangle.y + extentRectangle.height), (extentRectangle.x + extentRectangle.width), extentRectangle.y);
+                break;
+            case 1: // Started
+                cancel();
+                break;
+            case 2: // Paused
+                break;
+            case 3: // Succeeded
+                reset();
+                break;
+            case 4: // Failed
+                startExport(extentRectangle.x, (extentRectangle.y + extentRectangle.height), (extentRectangle.x + extentRectangle.width), extentRectangle.y);
+                break;
+            case 5: // Cancelling
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    function startExport(xSW, ySW, xNE, yNE) {
+        const vectorTiledLayer = map.basemap.baseLayers.get(0);
+        if (vectorTiledLayer.layerType !== Enums.LayerTypeArcGISVectorTiledLayer)
+            return;
+
+        // Create an envelope from the QML rectangle corners
+        const pointSW = mapView.screenToLocation(xSW, ySW);
+        const pointNE = mapView.screenToLocation(xNE, yNE);
+        const extent = ArcGISRuntimeEnvironment.createObject("Envelope", {
+            xMin: pointSW.x,
+            xMax: pointNE.x,
+            yMin: pointNE.y,
+            yMax: pointSW.y
+        });
+        const exportArea = GeometryEngine.normalizeCentralMeridian(GeometryEngine.project(extent, vectorTiledLayer.spatialReference));
+        extentRectangle.visible = false;
+        exportAreaGraphic.geometry = exportArea;
+
+        exportVectorTilesTask.url = vectorTiledLayer.url;
```

Comment:
I believe if `exportVectorTilesTask` is done loading, we can't change the url so we should add a check here but I could be mistaken